### DISCUSSION TITLE
Background

### DISCUSSION THREAD

**Discussion by MorreMor** [Clooos/Bubble-Card#1016] [Questions about config, custom styles and templates] [answered]
since update 2.3 background is black when I open a popup. before I had my background from my dashboard and now its black.... no background only black color

**Comment by Clooos** [comment]
Hi! Can you send me the YAML of your first pop-up on your view?

**Comment by MorreMor** [comment]
type: vertical-stack
cards:
  - type: custom:bubble-card
    card_type: pop-up
    hash: "#schlafzimmer"
    name: Schlafzimmer
    icon: mdi:bed
    state: sensor.nspanel_gartenbuero_temperature
    bg_blur: "20"
    shadow_opacity: "20"
    bg_opacity: "60"
    styles: ".icon-container {background-color:#453f3b!important;} "
    margin: 7px
    hide_backdrop: false
    button_type: state
    entity: sensor.klimaanlage_room_temperature_2
  - type: grid
    columns: 1
    square: false
    cards:
      - type: custom:bubble-card
        card_type: separator
        name: Lichter
        icon: mdi:led-strip
        styles: |+
          .separator-container div:last-child {
            opacity: 1!important;
           background-color: white !important;
          }

      - type: grid
        columns: 2
        square: false
        cards:
          - type: custom:bubble-card
            card_type: button
            entity: light.licht_schlafzimmer
            name: Lampe
            icon: mdi:ceiling-light
          - type: custom:bubble-card
            card_type: button
            entity: switch.licht_schrank
            name: Schrank
            icon: mdi:led-strip-variant
      - type: custom:gap-card
        height: 20
      - type: custom:bubble-card
        card_type: separator
        name: Klimaanlage
        icon: mdi:air-conditioner
        styles: |

          .bubble-line {
            background: white; 
           opacity: 1!important;
          }
        sub_button:
          - entity: sensor.klimaanlage_schlafzimmer_switch_0_power
            icon: null
            show_state: true
            show_background: false
      - features:
          - type: target-temperature
          - type: climate-hvac-modes
            hvac_modes:
              - cool
              - heat
              - dry
              - fan_only
              - "off"
          - type: climate-fan-modes
            style: dropdown
        type: tile
        entity: climate.klimaanlage_4
        name: Klima Schlafzimmer
        layout_options:
          grid_columns: 4
          grid_rows: 3
      - type: horizontal-stack
        cards:
          - type: custom:button-card
            entity: input_boolean.klima_schlafzimmer_boost
            show_name: true
            name: Kühlen 16°
            show_icon: true
            size: 10%
            icon: mdi:snowflake
            styles:
              card:
                - box-shadow: 0px 0px 10px 2px var(--button-card-light-color)
                - height: 50px
                - border: 0px solid var(--primary-background-color)
                - background: |
                    [[[
                      // Check the state of the input_boolean
                      const boostState = states['input_boolean.klima_schlafzimmer_boost'].state;

                      // If the input_boolean is on, apply orange shadow
                      if (boostState === 'on') {
                        return '0px 0px 10px 2px orange'; 
                      }

                      // If the input_boolean is off, apply white shadow
                      if (boostState === 'off') {
                        return '0px 0px 10px 2px '; // Apply white box shadow
                      }

                      return ''; // Default, no shadow
                    ]]]
            tap_action:
              action: toggle
          - type: custom:button-card
            entity: input_boolean.klima_schlafzimmer_normal
            show_name: true
            name: Kühlen 21°
            show_icon: true
            size: 10%
            icon: mdi:snowflake
            styles:
              card:
                - box-shadow: 0px 0px 10px 2px var(--button-card-light-color)
                - height: 50px
                - border: 0px solid var(--primary-background-color)
                - background: |
                    [[[
                      // Check the state of the input_boolean
                      const boostState = states['input_boolean.klima_schlafzimmer_normal'].state;

                      // If the input_boolean is on, apply orange shadow
                      if (boostState === 'on') {
                        return '0px 0px 10px 2px orange'; 
                      }

                      // If the input_boolean is off, apply white shadow
                      if (boostState === 'off') {
                        return '0px 0px 10px 2px '; // Apply white box shadow
                      }

                      return ''; // Default, no shadow
                    ]]]
            tap_action:
              action: toggle
      - type: custom:gap-card
        height: 20
      - type: custom:bubble-card
        card_type: separator
        name: Heizung
        icon: mdi:heating-coil
        styles: |+
          .separator-container div:last-child {
            opacity: 1!important;
          background-color: white !important;
          }

      - features:
          - type: target-temperature
          - type: climate-preset-modes
            preset_modes:
              - Aus
              - An
          - type: climate-fan-modes
            style: dropdown
        type: tile
        entity: climate.schlafzimmer_thermostat
        name: Schlafzimmer
        layout_options:
          grid_columns: 4
          grid_rows: 3
      - type: custom:gap-card
        height: 20
      - type: custom:bubble-card
        card_type: separator
        name: Rollladen
        icon: mdi:window-shutter
        styles: |
          .separator-container div:last-child {
            opacity: 1!important;
          background-color: white !important;
          }
  - type: custom:mushroom-cover-card
    name: Schlafzimmer
    icon: mdi:window-shutter
    entity: cover.rolladen_schlafzimmer
    layout: horizontal
    show_position_control: true
  - type: custom:gap-card
    height: 20
  - type: custom:bubble-card
    card_type: separator
    name: Sensors
    icon: mdi:motion-sensor
    styles: |
      .separator-container div:last-child {
        opacity: 1!important;
      background-color: white !important;
      }
  - type: custom:mushroom-template-card
    primary: Klimaanlage
    secondary: >-
      {{ states('sensor.klimaanlage_schlafzimmer_switch_0_power') | round(1) }}
      Watt
    icon: mdi:air-conditioner
  - type: custom:mushroom-template-card
    primary: Tablette
    secondary: "{{ states('sensor.tablette_heute_genommen')  }}"
    icon: mdi:pill
    tap_action:
      action: navigate
      navigation_path: "#tablette"

**Comment by Clooos** [comment]
Thank you! Can you also tell me what theme your using?

**Comment by scargill** [comment]
I've always used card-mod to put in backgrounds - doesn't work here so any tips I'm interested in too.

**Comment by MorreMor** [comment]
IOS Dark Mode blue red. I installed the old version and it works, don't now what's the prob. here are two screen shots with old and new version.
![IMG_0877](https://github.com/user-attachments/assets/e97f36f6-0fdc-4a94-8bc2-65ff47eff799)
![IMG_0878](https://github.com/user-attachments/assets/17903b69-45da-4a26-9b92-d7b9dba25156)

**Comment by Clooos** [comment]
Have you tried to lower `bg_opacity`? The backdrop behind the pop-ups is now darker so that's normal that your pop-up looks darker as well. But I will add an option to change the darkness of the backdrop, this way it will be possible to change it like it was before v2.3.0.

**Comment by anthonymkz** [comment, marked as answer]
> Have you tried to lower `bg_opacity`? The backdrop behind the pop-ups is now darker so that's normal that your pop-up looks darker as well. But I will add an option to change the darkness of the backdrop, this way it will be possible to change it like it was before v2.3.0.

I was about to submit my own issue report, but I think this issue may be related. I've also found the possible reason / fix. It seems some changes that are set to variables in the default styles may not behave correctly in iOS, particularly if the variable set in the styles is not defined in the theme being used.

In my instance, this manifested as changes to the blur amount working on a desktop browser but not translating to the iOS app. I was able to fix this by adding the -webkit-backdrop-filter style element to .bubble-pop-up for the card as seen below:
```
.bubble-pop-up {
  border-radius: 0px !important;
  -webkit-backdrop-filter: blur(10px) !important;
}
```
I don't know for a fact that this has to do with the variable not being defined in my theme but I can only assume that's the case. Hope this helps!

**Comment by Clooos** [comment]
Interesting, I will check that 👌 Thanks for the feedback!

**Comment by Lucaspog** [comment]
Problem with the button background. The black background keeps showing through.

The button is transparent with the following code:
![image](https://github.com/user-attachments/assets/8f7387b0-4e41-4919-88c9-ab8a84299e6c)
```
styles: |
  .bubble-button-card-container {
    background: rgba(0, 0, 0, 0.35) !important;
  }
```

But when choosing colors, using transparency makes the button black. The black background keeps showing through, which I can't remove black background .

There is already a problem here:
![image](https://github.com/user-attachments/assets/9e779886-9953-473e-8a72-495d70441866)
```
styles: |
  .bubble-button-background {
    opacity: 1 !important;
    background-color: ${state === 'on' ? 'blue' : 'rgba(0, 0, 0, 0.2)'} !important;
```

**Comment by anthonymkz** [comment]
> There is already a problem here: ![image](https://private-user-images.githubusercontent.com/122601994/391893907-9e779886-9953-473e-8a72-495d70441866.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNTY1OTcsIm5iZiI6MTczMzI1NjI5NywicGF0aCI6Ii8xMjI2MDE5OTQvMzkxODkzOTA3LTllNzc5ODg2LTk5NTMtNDczZS04YTcyLTQ5NWQ3MDQ0MTg2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwM1QyMDA0NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMWM0YjlhYjUzMmM1MzYwYjA0YjQ0MDQ0MjBmYzg0MzE0MGZlODBiOWMwMTVlMjk3OTA1MjQ0NWI1YjllMGE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Fb3acw2bklikZJA-dW7ytpburHCdf-lHkoVkQZxynp8)
> 
> ```
> styles: |
>   .bubble-button-background {
>     opacity: 1 !important;
>     background-color: ${state === 'on' ? 'blue' : 'rgba(0, 0, 0, 0.2)'} !important;
> ```

You need to set the opacity on the container and background to get the desired result. You are also creating an issue by setting the opacity to 1. Not sure why that's there. The following gets your desired result:

```
  .bubble-button-card-container {
    background-color: ${state === 'on' ? 'blue' : 'rgba(0, 0, 0, 0.2)'} !important;
  }

  .bubble-button-background {
    background-color: ${state === 'on' ? 'blue' : 'rgba(0, 0, 0, 0.2)'} !important;
  }
```

**Comment by Lucaspog** [comment]
Thank you very much for your help. Now the style works correctly, my code for transparent buttons.

```
styles: |
  .bubble-button-card-container {
    background-color: ${state === 'on' ? 'rgba(255, 157, 10, 0.5)' : 'rgba(0, 0, 0, 0.2)'} !important;
  }

  .bubble-button-background {
    background-color: ${state === 'on' ? 'rgba(255, 157, 10, 0.5)' : 'rgba(0, 0, 0, 0.2)'} !important;
  }

  .bubble-icon-container {
    background: ${state === 'on' ? 'rgba(62, 58, 52, 0.8)' : 'rgba(255, 255, 255, 0.1)'} !important;
  }
  .bubble-sub-button-1 {
    background-color: ${state === 'on' ? 'rgba(62, 58, 52, 0.8)' : 'rgba(255, 255, 255, 0.1)'} !important;
  }
```
